site/src/pages/CreateWorkspacePage/CreateWorkspacePage.tsx

```diff
@@ -1,62 +1,155 @@
+  const [authState] = useActor(xServices.authXService)
+  const { me } = authState.context
   const featureVisibility = useSelector(
     xServices.entitlementsXService,
     selectFeatureVisibility,
     shallowEqual,
   )
   const workspaceQuotaEnabled = featureVisibility[FeatureNames.WorkspaceQuota]
 
-  const [authState] = useActor(xServices.authXService)
-  const { me } = authState.context
-  const [createWorkspaceState, send] = useMachine(createWorkspaceMachine, {
-    context: { organizationId, templateName, workspaceQuotaEnabled, owner: me ?? null },
-    actions: {
-      onCreateWorkspace: (_, event) => {
-        navigate(`/@${event.data.owner_name}/${event.data.name}`)
+  const [owner, setOwner] = useState<User | null>(me ?? null)
+  const [templates, setTemplates] = useState<Template[]>()
+  const [selectedTemplate, setSelectedTemplate] = useState<Template>()
+  const [templateSchema, setTemplateSchema] = useState<ParameterSchema[]>()
+  const [creatingWorkspace, setCreatingWorkspace] = useState<boolean>(false)
+  const [createWorkspaceError, setCreateWorkspaceError] = useState<Error | unknown>()
+  const [getTemplatesError, setGetTemplatesError] = useState<Error | unknown>()
+  const [getTemplateSchemaError, setGetTemplateSchemaError] = useState<Error | unknown>()
+  const [permissions, setPermissions] = useState<Record<string, boolean>>()
+  const [checkPermissionsError, setCheckPermissionsError] = useState<Error | unknown>()
+  const [workspaceQuota, setWorkspaceQuota] = useState<WorkspaceQuota>()
+  const [getWorkspaceQuotaError, setGetWorkspaceQuotaError] = useState<Error | unknown>()
+
+  useEffect(() => {
+    setGetTemplatesError(undefined)
+    getTemplates(organizationId).then(
+      (res) => {
+        const temps = res.filter((template) => template.name === templateName)
+        const selectedTemps = res.length > 0 ? temps[0] : undefined
+        setSelectedTemplate(selectedTemps)
+        setTemplates(temps)
+      },
+      (err) => {
+        setGetTemplatesError(err)
+      },
+    )
+  }, [organizationId, templateName])
+
+  useEffect(() => {
+    if (!selectedTemplate) {
+      return
+    }
+
+    setGetTemplateSchemaError(undefined)
+    getTemplateVersionSchema(selectedTemplate.active_version_id).then(
+      (res) => {
+        // Only show parameters that are allowed to be overridden.
+        // CLI code: https://github.com/coder/coder/blob/main/cli/create.go#L152-L155
+        res = res.filter((param) => param.allow_override_source)
+        setTemplateSchema(res)
+      },
+      (err) => {
+        setGetTemplateSchemaError(err)
       },
-    },
-  })
+    )
+  }, [selectedTemplate])
 
-  const {
-    templates,
-    templateSchema,
-    selectedTemplate,
-    getTemplateSchemaError,
-    getTemplatesError,
-    createWorkspaceError,
-    permissions,
-    workspaceQuota,
-    getWorkspaceQuotaError,
-    owner,
-  } = createWorkspaceState.context
+  useEffect(() => {
```

Comment:
Keep in mind that `useEffects` are really meant to handle side effects. It's best to use them sparingly and keep them pretty lean. I really like that you have multiple `useEffects` here broken up by concern - that's definitely the right way to think about them.